### PR TITLE
Increased Coverage of LockExtension

### PR DESCRIPTION
**Pull request by jordisala1991**
### Subject

I'm trying to increase the coverage of the project. I just picked up the first class on the first folder. I will try to create a PR for every class that I cover.

Have some questions for the contributors:

- Do you like the old getMock method from PHPUnit? (will be deprecated in future versions of PHPUnit). Personally I think [prophecy](https://github.com/phpspec/prophecy) is better and cleaner, but I saw everything using getMock, so didn't want to change it at first.

- What do you think about using Phpdoc to mark a function as a test instead of having it as a name.

### To do

- [x] Update the tests

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,6 +8,7 @@ composer.lock
 /vendor
 composer.phar
 .php_cs.cache
+.DS_Store
```
Comment (Soullivaneuh) [inline .gitignore]: This have nothing to do on project `.gitignore`.

This file is OS specific. See: https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

File: Tests/Admin/Extension/LockExtensionTest.php [hunk 2]
```diff
@@ -32,86 +32,201 @@ class LockExtensionTest extends \PHPUnit_Framework_TestCase
      */
     private $eventDispatcher;
 
+    /**
+     * @var AbstractAdmin
```
Comment (Soullivaneuh) [inline Tests/Admin/Extension/LockExtensionTest.php]: `AdminInterface`

**Comment by greg0ire** [comment]
> Do you like the old getMock method from PHPUnit? (will be deprecated in future versions of PHPUnit). Personally I think prophecy is better and cleaner, but I saw everything using getMock, so didn't want to change it at first.

I like prophecy better, and am already using it in several tests. Don't hesitate to use it, it is way more readable.

**Comment by greg0ire** [comment]
Please squash some or all of the commits together

**Comment by Soullivaneuh** [comment]
> Do you like the old getMock method from PHPUnit? (will be deprecated in future versions of PHPUnit).

Do you have a source of what you are saying?

File: Tests/Admin/Extension/LockExtensionTest.php [hunk 3]
```diff
@@ -32,86 +32,201 @@ class LockExtensionTest extends \PHPUnit_Framework_TestCase
...
 
+    /**
+     * @var LockInterface
+     */
     private $modelManager;
 
-    private $form;
-
+    /**
+     * @var StdClass
```
Comment (OskarStark) [inline Tests/Admin/Extension/LockExtensionTest.php]: `stdClass` instead of `StdClass` imo

**Comment by OskarStark** [comment]
// cc @ahmetakbn for your information ;-)

File: Tests/Admin/Extension/LockExtensionTest.php (hunk 3 above)
Comment (core23) [inline Tests/Admin/Extension/LockExtensionTest.php]: `\stdClass`

**Comment by jordisala1991** [comment]
@Soullivaneuh https://github.com/sebastianbergmann/phpunit/commit/5b6283a80cdb98dd482a86eee75f8e6a8f598cd8#diff-511c90242d5589e941582006e0485184R1445 This is for PHPUnit 5.4

@greg0ire I prefer using the phpdoc for 2 reasons:

- More flexibility with the test name and less repetition of the word test.
- With the phpdoc you can easily see what is a test and what is not. For example in the Test class of the PR there are 3 private methods in order to reuse mock objects but is hard to distinguish it from the actual tests.

**Comment by greg0ire** [comment]
@jordisala1991 : would you please open issues / PRs on [Sonata dev-kit](https://github.com/sonata-project/dev-kit) to submit these recommendations to a vote? I think both would be great, but it would be best to document decisions we make, why we make them, and how new tests are supposed to be written (and how old tests are supposed to evolve, ideally).

**Comment by Soullivaneuh** [comment]
> @Soullivaneuh sebastianbergmann/phpunit@5b6283a#diff-511c90242d5589e941582006e0485184R1445 

This is a deprecation for `getMock` only.

> This is for PHPUnit 5.4. And getMockBuilder is even worse...

Arbitrary position. :smile:

IMHO, actual tests are using the mock system and should be kept for consistency.

Maybe another discussion can be launched and dev-kit to change that.

File: Tests/Admin/Extension/LockExtensionTest.php [hunk 4]
```diff
@@ -32,86 +32,201 @@ class LockExtensionTest extends \PHPUnit_Framework_TestCase
...
     public function setUp()
     {
-        $contractor = $this->getMock('Sonata\AdminBundle\Builder\FormContractorInterface');
-
-        $formFactory = $this->getMock('Symfony\Component\Form\FormFactoryInterface');
         $this->eventDispatcher = new EventDispatcher();
-
-        $formBuilder = new FormBuilder('form', null, $this->eventDispatcher, $formFactory);
+        $this->request = new Request();
+        $this->object = new \StdClass();
```
Comment (core23) [inline Tests/Admin/Extension/LockExtensionTest.php]: `\stdClass` please

**Comment by jordisala1991** [comment]
Fixed the commits and squashed.

**Comment by OskarStark** [comment]
LGTM

ping @greg0ire @Soullivaneuh

**Pull request merged**